nimterop/types.nim

```diff
@@ -31,3 +31,31 @@ when defined(c):
 elif defined(cpp):
   type
     wchar_t* {.importc.} = object
+
+template enumOp*(op, typ, typout) =
+  proc op*(x: typ, y: int): typout {.borrow.}
+  proc op*(x: int, y: typ): typout {.borrow.}
+  proc op*(x, y: typ): typout {.borrow.}
+
+template defineEnum*(typ: untyped) =
+  type
+    typ* = distinct int
+
+  enumOp(`+`,   typ, typ)
+  enumOp(`-`,   typ, typ)
+  enumOp(`*`,   typ, typ)
+  enumOp(`<`,   typ, bool)
+  enumOp(`<=`,  typ, bool)
+  enumOp(`==`,  typ, bool)
+  enumOp(`div`, typ, typ)
+  enumOp(`mod`, typ, typ)
+
+  proc `shl`*(x: typ, y: int): typ {.borrow.}
+  proc `shl`*(x: int, y: typ): typ {.borrow.}
+  proc `shl`*(x, y: typ): typ {.borrow.}
+
+  proc `shr`*(x: typ, y: int): typ {.borrow.}
+  proc `shr`*(x: int, y: typ): typ {.borrow.}
+  proc `shr`*(x, y: typ): typ {.borrow.}
+
+  proc `$` *(x: typ): string {.borrow.}
```

Comment:
can't we instead use enum element name in source code? treesitter shd expose it, eg:
`$blue => "blue" instead of 23`